typed-racket-lib/typed-racket/types/subtype.rkt

```diff
@@ -45,9 +48,8 @@
   (and (subval* (seen) v1 v2) #t))
 
 ;; are t1 and t2 equivalent types (w.r.t. subtyping)
-(define (type-compare? t1 t2)
-  (or (equal? t1 t2) (and (subtype t1 t2)
-                          (subtype t2 t1))))
+(define (type-equiv? t1 t2)
+  (type≡? (seen) t1 t2))
```

Comment:
That every call to this dereferences a parameter is not great. We should try to fix this somehow later.